pkg/controller/service_lb.go

```diff
@@ -0,0 +1,396 @@
+package controller
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	netattachdef "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+
+	"github.com/kubeovn/kube-ovn/pkg/util"
+)
+
+const (
+	LbPodImage       = "kubeovn/vpc-nat-gateway:v1.11.0"
+	INIT_ROUTE_TABLE = "init"
+	POD_EIP_ADD      = "eip-add"
+	POD_DNAT_ADD     = "dnat-add"
+)
+
+func genLbSvcDpName(name string) string {
+	return fmt.Sprintf("lb-svc-%s", name)
+}
+
+func (c *Controller) applyExternalNetwork(svc *corev1.Service) error {
+	nic := "eth0"
```

Comment:
nic should be configurable

Reply:
The annotation "ovn.kubernetes.io/attchmentnic" is used to config nic, the eth0 is used as default value